enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/analytics/AnalyticsFilter.java

```diff
@@ -212,6 +164,80 @@ public void handleFailureRequest(RequestContext requestContext) {
         }
     }
 
+    private void addInsightsMetaData(RequestContext requestContext) {
+        String apiName = requestContext.getMatchedAPI().getName();
+        String apiVersion = requestContext.getMatchedAPI().getVersion();
+        String apiType = requestContext.getMatchedAPI().getApiType();
+        String projectId = requestContext.getMatchedAPI().getChoreoComponentInfo().getProjectID();
```

Comment:
I guess we shoud check the possibility of this property being null and assign some default value.